iris-mpc-gpu/src/server/actor.rs

```diff
@@ -652,6 +652,23 @@ impl ServerActor {
             );
         }
 
+        ///////////////////////////////////////////////////////////////////
+        // SYNC BATCH CONTENTS AND FILTER OUT INVALID ENTRIES
+        ///////////////////////////////////////////////////////////////////
+        let tmp_now = Instant::now();
+        tracing::info!("Syncing batch entries");
+
+        // Compute hash of the SNS message ids concatenated
+        let batch_hash = sha256_bytes(batch.sns_message_ids.join(""));
+        tracing::info!("Current batch hash: {}", hex::encode(&batch_hash[0..4]));
+
+        let valid_entries =
+            self.sync_batch_entries(&batch.valid_entries, self.max_batch_size, &batch_hash)?;
+        let valid_entry_idxs = valid_entries.iter().positions(|&x| x).collect::<Vec<_>>();
+        batch_size = valid_entry_idxs.len();
+        batch.retain(&valid_entry_idxs);
```

Comment:
You should also remove those `sns_message_ids` from here to be consistent: https://github.com/worldcoin/iris-mpc/blob/e85d8d4c52a96302fb947708a1ef629b696ab8e6/iris-mpc-gpu/src/server/mod.rs#L174